api/envoy/config/filter/http/ext_authz/v2/ext_authz.proto

```diff
@@ -72,6 +72,20 @@ message ExtAuthz {
   // Sets the HTTP status that is returned to the client when there is a network error between the
   // filter and the authorization server. The default status is HTTP 403 Forbidden.
   envoy.type.HttpStatus status_on_error = 7;
+
+  // Specifies a list of metadata namespaces whose values, if present, will be passed to the
```

Comment:
As API shepherd feedback, I'm wondering what the tradeoff here is between supported dynamic metadata and typed filter state? I thought the latter was plan-of-record going forward.

Reply:
See also https://github.com/envoyproxy/envoy/issues/4929

Reply:
I have no opinion on the matter. However, for my specific use-case the existing `jwt_authn` filter propagates its metadata via dynamic metadata and not filter state. If we convert that to use filter state (and we have some story about passing filter state to an external system like ext_authz), then i'm happy to rework this to use that. Though from the issue you linked, it looks like there are still some open questions.

I don't want to be creating more broken windows if I can help it. Does this look like it will cause a significant amount of tech debt?

Reply:
Given the lack of serializabiltiy of `FilterState`, I think we have to go with dynamic metadata, so happy to put this thread aside for now.